src/Symfony/Component/Translation/DataCollectorTranslator.php

```diff
@@ -56,11 +59,18 @@ public function trans($id, array $parameters = array(), $domain = null, $locale
 
     /**
      * {@inheritdoc}
+     *
+     * @deprecated since Symfony 4.2, use the trans() method instead with a %count% parameter
      */
     public function transChoice($id, $number, array $parameters = array(), $domain = null, $locale = null)
     {
+        if ($this->translator instanceof TranslatorInterface) {
+            $trans = $this->translator->trans($id, array('%count%' => $number) + $parameters, $domain, $locale);
+        }
+
         $trans = $this->translator->transChoice($id, $number, $parameters, $domain, $locale);
-        $this->collectMessage($locale, $domain, $id, $trans, $parameters, $number);
+
+        $this->collectMessage($locale, $domain, $id, $trans, array('%count%' => $number) + $parameters);
```

Comment:
I've always got problems when adding arrays. Shouldn't we use the more easy-to-read array_merge?

Reply:
array_merge is a function call, personally I prefer using the language directly.